python/ray/data/_internal/execution/interfaces/execution_options.py

```diff
@@ -183,10 +223,12 @@ class ExecutionOptions:
             option is useful for performance debugging. On by default.
     """
 
-    resource_limits: ExecutionResources = field(default_factory=ExecutionResources)
+    _resource_limits: ExecutionResources = field(
```

Comment:
Do we need to worry about users (or us) specifying `resource_limits` in the `ExecutionOptions` constructor? Currently, it'll error.

```python
>>> ExecutionOptions(resource_limits=...)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ExecutionOptions.__init__() got an unexpected keyword argument 'resource_limits'
```

Reply:
Good point. there are indeed some test failures because of this.

Reply:
fixed